File: frontend/src/Chapter/ChapterRelease.ts
// Sonarr divergence: NEW manga-domain typing per Phase 16 STRUCT-09 — see DIVERGENCE.md.
// TV's frontend has no per-release type because language lives on EpisodeFile.Languages.
// Manga's per-translation grain demands a nested collection — multilingual scanlations
// exist for manga but not for TV (Phase 16 STRUCT-02 + CONTEXT D-01).
// Mirrors the API V5 ChapterReleaseResource wire shape (Plan 16-05).
//
// Property semantics (Sonarr-divergence note):
//   * id                 — primary key of the underlying ChapterRelease row.
//   * translatedLanguage — BCP-47 language tag (e.g. 'en', 'es', 'ja').
//   * scanlationGroup    — translator group / publisher; optional.
//   * releaseDate        — ISO 8601 per-translation upload time. Distinct from
//                          Chapter.firstReleaseDate (chapter-publish date —
//                          Sonarr-mirror of Episode.airDateUtc).
//   * externalId         — upstream source identifier (e.g. MangaDex chapter UUID).
//
// Phase 16 D-04 contract: a Chapter with `releases.length === 0` renders as
// "Missing" (Sonarr-mirror of Episode + AirDateUtc + no EpisodeFile pattern).
export interface ChapterRelease {
  id: number;
  translatedLanguage: string;
  scanlationGroup?: string;
  releaseDate?: string;     // ISO 8601 — per-translation upload time
  externalId?: string;
}

export default ChapterRelease;

File: src/NzbDrone.Core/Manga/ChapterRelease.cs
using System;
using NzbDrone.Core.Datastore;

namespace NzbDrone.Core.Manga
{
    // Sonarr divergence: NEW manga sibling per Phase 16 STRUCT-02 — see DIVERGENCE.md.
    // Captures the conscious manga-domain divergence from Sonarr's Episode shape:
    // multilingual scanlations exist for manga but not for TV. Per Phase 16
    // CONTEXT D-01 (stale-release retention) + D-02 (FirstReleaseDate denormalization),
    // this entity carries per-(language, group) translation upload metadata while
    // the canonical Chapter row stays language-free at the (MangaId, ChapterNumber) grain.
    // Role-match analog: src/NzbDrone.Core/Manga/Chapter.cs (sibling on same ModelBase).
    public class ChapterRelease : ModelBase
    {
        public int ChapterId { get; set; }                    // soft FK to Chapters.Id; cascade in C#
        public string TranslatedLanguage { get; set; }        // BCP-47, NotNullable
        public string ScanlationGroup { get; set; }           // Nullable
        public DateTime? ReleaseDate { get; set; }            // per-translation upload time (NOT chapter-publish date — D-02)
        public string ExternalId { get; set; }                // Nullable (e.g. MangaDex chapter UUID)
    }
}
